secure/secure.sh

```diff
@@ -0,0 +1,403 @@
+  set_property_yarn_site 'yarn.nodemanager.linux-container-executor.group' 'yarn'
+
+  # HTTPS
+  set_property_yarn_site 'yarn.http.policy' 'HTTPS_ONLY'
+  set_property_yarn_site 'yarn.log.server.url' "https://$MASTER_FQDN:19889/jobhistory/logs"
+
+  cat << EOF > $HADOOP_CONF_DIR/container-executor.cfg
+yarn.nodemanager.linux-container-executor.group=yarn
+banned.users=hdfs,yarn,mapred,bin
+min.user.id=1000
+EOF
+}
+
+function set_property_mapred_site() {
+  set_property_in_xml $HADOOP_CONF_DIR/mapred-site.xml $1 $2
+}
+
+function config_mapred_site() {
+  if [[ $ROLE == 'Master' ]]; then
+    set_property_mapred_site 'mapreduce.jobhistory.keytab' "$HADOOP_CONF_DIR/mapred.keytab"
+    set_property_mapred_site 'mapreduce.jobhistory.principal' "mapred/_HOST@$REALM"
+  fi
+
+  set_property_mapred_site 'mapreduce.jobhistory.http.policy' 'HTTPS_ONLY'
+  set_property_mapred_site 'mapreduce.ssl.enabled' 'true'
+  set_property_mapred_site 'mapreduce.shuffle.ssl.enabled' 'true'
+  set_property_mapred_site 'mapreduce.jobhistory.address' "$MASTER_FQDN:10020"
+  set_property_mapred_site 'mapreduce.jobhistory.webapp.https.address' "$MASTER_FQDN:19889"
+}
+
+function copy_keystore_files() {
+  mkdir $HADOOP_CONF_DIR/ssl
+  gsutil cp $keystore_uri $HADOOP_CONF_DIR/ssl/keystore.jks
+  gsutil cp $truststore_uri $HADOOP_CONF_DIR/ssl/truststore.jks
+  chown -R yarn:hadoop $HADOOP_CONF_DIR/ssl
+  chmod 755 $HADOOP_CONF_DIR/ssl
+  chmod 440 $HADOOP_CONF_DIR/ssl/keystore.jks
+  chmod 444 $HADOOP_CONF_DIR/ssl/truststore.jks
+}
+
+function set_property_ssl_xml() {
+  set_property_in_xml $HADOOP_CONF_DIR/ssl-$1.xml $2 $3
+}
+
+function config_ssl_xml() {
+  cp $HADOOP_CONF_DIR/ssl-$1.xml.example $HADOOP_CONF_DIR/ssl-$1.xml
+  set_property_ssl_xml $1 "ssl.$1.truststore.location" "$HADOOP_CONF_DIR/ssl/truststore.jks"
+  set_property_ssl_xml $1 "ssl.$1.truststore.password" "$keystore_password"
+  set_property_ssl_xml $1 "ssl.$1.truststore.type" "jks"
+  set_property_ssl_xml $1 "ssl.$1.keystore.location" "$HADOOP_CONF_DIR/ssl/keystore.jks"
+  set_property_ssl_xml $1 "ssl.$1.keystore.password" "$keystore_password"
+  set_property_ssl_xml $1 "ssl.$1.keystore.keypassword" "$keystore_password"
+  set_property_ssl_xml $1 "ssl.$1.keystore.type" "jks"
+}
+
+function set_property_hive_site() {
+  set_property_in_xml "/etc/hive/conf/hive-site.xml" $1 $2
+}
+
+function secure_hive() {
+  if [[ $ROLE == 'Master' ]]; then
+    kadmin -p root -w $root_principal_password -q "addprinc -randkey hive/$FQDN"
+    kadmin -p root -w $root_principal_password -q "xst -k hive.keytab hive/$FQDN"
+    mv ./hive.keytab /etc/hive/conf/
+    chown hive:hive /etc/hive/conf/hive.keytab
+    chmod 400 /etc/hive/conf/hive.keytab
+    # Add the user 'hive' to group 'hadoop' so it can access the keystore files
+    usermod -a -G hadoop hive
+    # HiveServer2
+    set_property_hive_site 'hive.server2.authentication' 'KERBEROS'
```

Comment:
Does `hive.server2.authentication` need to be set on workers as well so that you can run `beeline` on a worker node?

Reply:
Done.